lib/kafka/produce_operation.rb

```diff
@@ -137,6 +137,9 @@ def handle_response(response)
               delay: ack_time - message.create_time,
             })
           end
+        rescue Kafka::MessageSizeTooLarge
+          @logger.error "Message is too large when writing to #{topic}/#{partition}"
+          @buffer.clear_messages(topic: topic, partition: partition)
```

Comment:
I'm not sure I've ever seen CorruptMessage in the wild, though – I imagine you'd incur the same message loss since the producer wouldn't be able to deliver the buffered messages to the partition, and forcibly restarting it would clear those messages anyway.

Unfortunately, the Kafka protocol doesn't indicate _which_ message caused the problem, making it rather difficult to only clear the problematic one.

One option would be to go through each buffered message for the partition and remove the _largest one_. But it's sort of a hack...